### PR TITLE
Add GitHub Action to publish to GHCR

### PR DESCRIPTION
**Pull request by joshbuker**

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+name: Docker
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  schedule:
+    - cron: '44 6 * * *'
```
Comment (eamonburns) [inline .github/workflows/docker-publish.yml]: Is this 6:44 every day? Why does it need to be every day?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Extract metadata (tags, labels) for Docker
+      # https://github.com/docker/metadata-action
+      - name: Extract Docker metadata
```
Comment (eamonburns) [inline .github/workflows/docker-publish.yml:66]: Are the tags extracted from the Git tags? For instance, if I create a `v0.1.0` Git tag, will a Docker image be pushed with that tag as well?

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (joshbuker) [reply .github/workflows/docker-publish.yml]: It doesn't, necessarily. Although daily builds can be a good idea to catch issues. I just copied the defaults for this, can comment out the cron schedule.

File: .github/workflows/docker-publish.yml (hunk 2 above)
Comment (joshbuker) [reply .github/workflows/docker-publish.yml:66]: I believe so, yes. I haven't tried it yet.

Comment (eamonburns) [reply .github/workflows/docker-publish.yml:66]: Ok, I looked at the documentation, and it looks like there is [a type specifically for version tags](https://github.com/docker/metadata-action?tab=readme-ov-file#typesemver) which is not in [the defaults](https://github.com/docker/metadata-action?tab=readme-ov-file#tags-input). The default just creates a docker image tag for every git tag that is pushed (`type=ref,event=tag`). I think the default is good for now though

**Pull request merged**